File: AmiClean-Backend/AmiClean.Domain/Entities/Recenzija.cs
namespace AmiClean.Domain.Entities;

public class Recenzija
{
    public int ID_Recenzije { get; set; }
    public int FK_Korisnik { get; set; }
    public int FK_Narudzba { get; set; }
    public int Ocjena { get; set; }
    public string? Komentar { get; set; }
    public DateTime Datum_Objave { get; set; }

    public virtual Korisnik Korisnik { get; set; } = null!;
    public virtual Narudzba Narudzba { get; set; } = null!;
}

File: AmiClean-App/lib/features/recenzije/models/recenzija.dart
class Recenzija {
  const Recenzija({
    required this.id,
    required this.narudzbaId,
    required this.ocjena,
    this.komentar,
    required this.datumObjave,
  });

  final int id;
  final int narudzbaId;
  final int ocjena;
  final String? komentar;
  final DateTime datumObjave;

  factory Recenzija.fromJson(Map<String, dynamic> json) {
    return Recenzija(
      id: json['id'] as int,
      narudzbaId: json['narudzbaId'] as int,
      ocjena: json['ocjena'] as int,
      komentar: json['komentar'] as String?,
      datumObjave: DateTime.parse(json['datumObjave'] as String),
    );
  }
}
